frontend/src/pages/admin/questions.tsx

```diff
@@ -0,0 +1,185 @@
+import { NextPage } from 'next'
+import { v4 as uuidv4 } from 'uuid'
+import React, { useState } from 'react'
+import { FiPlus } from 'react-icons/fi'
+import { useForm } from 'react-hook-form'
+import { GrFormSubtract } from 'react-icons/gr'
+
+import Layout from '~/layouts/adminLayout'
+import { ICategory, IChoice } from '~/data/interfaces'
+import { categories } from '~/data/categories'
+import { classNames } from '~/utils/classNames'
+import { QuestionFormValues } from '~/data/types'
+
+const Questions: NextPage = (): JSX.Element => {
+  /**
+    * This will partially save your choices data
+   */
+  const [choices, setChoices] = useState<IChoice[]>([
+    {
+      id: uuidv4(),
+      value: ''
+    }
+  ])
+  
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting, errors }
+  } = useForm<QuestionFormValues>()
+
+
+  /**
+    * This will handle dynamic input field
+   */
+  const handleChangeInput = (id: string, event: React.ChangeEvent<HTMLInputElement>) => {
+    const newInputFields = choices.map((i: any) => {
+      if (id === i.id) {
+        i[event.target.name] = event.target.value
+      }
+      return i
+    })
+    setChoices(newInputFields)
+  }
+
+  /**
+    * This will remove dynamic choice field
+   */
+  const handleRemoveField = (id: string) => {
+    const values = [...choices]
+    values.splice(
+      values.findIndex((value) => value.id === id),
+      1
+    )
+    setChoices(values)
+  }
+
+  /**
+    * This will add dynamic choice field
+   */
+  const handleAddField = (): void => setChoices([...choices, { id: uuidv4(), value: '' }])
+
+    /**
+    * This will Save all the data seelcted
+   */
+  const handleSave = (data: QuestionFormValues) => {
+    console.log('Category ID', data?.category_id)
+    console.log('Question: ', data?.value)
+    console.log('Choices: ', choices)
```

Comment:
please don't forget to remove when not in use anymore. for now it is ok because it is for checking